plugins/header_rewrite/header_rewrite_test.cc

```diff
@@ -31,274 +33,395 @@ const char PLUGIN_NAME_DBG[] = "TEST_dbg_header_rewrite";
     CHECK_EQ(p.getTokens()[2], " =<>\"#\\");
+
+    END_TEST();
   }
 
   {
     ParserTest p("add-header foo \\<bar\\>");
-    CHECK_EQ(p.getTokens().size(), 3);
+
+    CHECK_EQ(p.getTokens().size(), 3UL);
     CHECK_EQ(p.getTokens()[0], "add-header");
     CHECK_EQ(p.getTokens()[1], "foo");
     CHECK_EQ(p.getTokens()[2], "<bar>");
+
+    END_TEST();
   }
 
   {
     ParserTest p("add-header foo \\bar\\");
-    CHECK_EQ(p.getTokens().size(), 3);
+
+    CHECK_EQ(p.getTokens().size(), 3UL);
     CHECK_EQ(p.getTokens()[0], "add-header");
     CHECK_EQ(p.getTokens()[1], "foo");
     CHECK_EQ(p.getTokens()[2], "bar");
+
+    END_TEST();
   }
 
   {
     ParserTest p("add-header foo \"bar\"");
-    CHECK_EQ(p.getTokens().size(), 3);
+
+    CHECK_EQ(p.getTokens().size(), 3UL);
     CHECK_EQ(p.getTokens()[0], "add-header");
     CHECK_EQ(p.getTokens()[1], "foo");
     CHECK_EQ(p.getTokens()[2], "bar");
+
+    END_TEST();
   }
 
   {
     ParserTest p("add-header foo \"\\\"bar\\\"\"");
-    CHECK_EQ(p.getTokens().size(), 3);
+
+    CHECK_EQ(p.getTokens().size(), 3UL);
     CHECK_EQ(p.getTokens()[0], "add-header");
     CHECK_EQ(p.getTokens()[1], "foo");
     CHECK_EQ(p.getTokens()[2], "\"bar\"");
+
+    END_TEST();
   }
 
   {
     ParserTest p("add-header foo \"\\\"\\\\\\\"bar\\\\\\\"\\\"\"");
-    CHECK_EQ(p.getTokens().size(), 3);
+
+    CHECK_EQ(p.getTokens().size(), 3UL);
     CHECK_EQ(p.getTokens()[0], "add-header");
     CHECK_EQ(p.getTokens()[1], "foo");
     CHECK_EQ(p.getTokens()[2], "\"\\\"bar\\\"\"");
+
+    END_TEST();
   }
 
   {
     ParserTest p("add-header Public-Key-Pins \"max-age=3000; pin-sha256=\\\"d6qzRu9zOECb90Uez27xWltNsj0e1Md7GkYYkVoZWmM=\\\"\"");
-    CHECK_EQ(p.getTokens().size(), 3);
+
+    CHECK_EQ(p.getTokens().size(), 3UL);
     CHECK_EQ(p.getTokens()[0], "add-header");
     CHECK_EQ(p.getTokens()[1], "Public-Key-Pins");
     CHECK_EQ(p.getTokens()[2], "max-age=3000; pin-sha256=\"d6qzRu9zOECb90Uez27xWltNsj0e1Md7GkYYkVoZWmM=\"");
+
+    END_TEST();
   }
 
   {
-    ParserTest p(
-      "add-header Public-Key-Pins max-age\\=3000;\\ pin-sha256\\=\\\"d6qzRu9zOECb90Uez27xWltNsj0e1Md7GkYYkVoZWmM\\=\\\"");
-    CHECK_EQ(p.getTokens().size(), 3);
+    ParserTest p("add-header Public-Key-Pins max-age\\=3000;"
+                 "\\ pin-sha256\\=\\\"d6qzRu9zOECb90Uez27xWltNsj0e1Md7GkYYkVoZWmM\\=\\\"");
+
+    CHECK_EQ(p.getTokens().size(), 3UL);
     CHECK_EQ(p.getTokens()[0], "add-header");
     CHECK_EQ(p.getTokens()[1], "Public-Key-Pins");
     CHECK_EQ(p.getTokens()[2], "max-age=3000; pin-sha256=\"d6qzRu9zOECb90Uez27xWltNsj0e1Md7GkYYkVoZWmM=\"");
+
+    END_TEST();
+  }
+
+  {
+    ParserTest p("add-header Client-IP \"%<chi>\"");
+
+    CHECK_EQ(p.getTokens().size(), 3UL);
+    CHECK_EQ(p.getTokens()[0], "add-header");
+    CHECK_EQ(p.getTokens()[1], "Client-IP");
+    CHECK_EQ(p.getTokens()[2], "\%<chi>");
+
+    END_TEST();
+  }
+
+// This can not succeed as is now, have to quote the "%<proto>"
```

Comment:
Not sure I understand, this check breaks now? There is an incompatible change since 6.x, where just %<chi> no longer works. I can remove this check.